src/main/kotlin/me/serce/solidity/ide/formatting/SolidityExternalFormatter.kt

```diff
@@ -0,0 +1,126 @@
+package me.serce.solidity.ide.formatting
+
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.process.CapturingProcessAdapter
+import com.intellij.execution.process.OSProcessHandler
+import com.intellij.execution.process.ProcessEvent
+import com.intellij.formatting.FormattingContext
+import com.intellij.formatting.service.AsyncDocumentFormattingService
+import com.intellij.formatting.service.AsyncFormattingRequest
+import com.intellij.formatting.service.FormattingService.Feature
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import me.serce.solidity.lang.core.SolidityFile
+import me.serce.solidity.settings.FormatterType
+import me.serce.solidity.settings.SoliditySettings
+import java.nio.charset.StandardCharsets
+import java.util.*
+import java.util.concurrent.ExecutionException
+
+class SolidityExternalFormatter : AsyncDocumentFormattingService() {
+
+  override fun createFormattingTask(request: AsyncFormattingRequest): FormattingTask? {
+    val formattingContext: FormattingContext = request.context
+    val project = formattingContext.project
+    val settings = SoliditySettings.getInstance(project)
+    val userHome = System.getProperty("user.home")
+
+    if (settings.formatterType == FormatterType.INTELLIJ_SOLIDITY) {
+      return null
+    }
+
+    val psiFile: PsiFile = request.context.containingFile
+    val foundryExePath = settings.executablePath.ifBlank { "$userHome/.foundry/bin/forge" }
```

Comment:
👀 I think the default path wouln't work on Windows, let me also try to address this in a separate PR.